pkg/services/annotations/annotationsimpl/composite_store_test.go

```diff
@@ -122,16 +123,40 @@ func TestCompositeStore(t *testing.T) {
 			{Tag: "key2:val2"},
 		}
 
-		res, _ := store.GetTags(context.Background(), nil)
+		res, _ := store.GetTags(context.Background(), annotations.TagsQuery{})
 		require.Equal(t, expected, res.Tags)
 	})
+
+	// Check if reader is not modifying query since it might cause a race condition in case of composite store
+	t.Run("should not modify query", func(t *testing.T) {
```

Comment:
we should probably rename the test, in so that we need to fix this long term, about modifying the query. this test stays until fixed? or to create an issue so that we do not lose this at some point